src/azure_cosmosdb/cosmosdb_data_mappings.bal

```diff
@@ -0,0 +1,486 @@
+    return params;
+}
+
+isolated function mapOfferHeaderType(string httpVerb, string url) returns HeaderParameters {
+    HeaderParameters params = {};
+    params.verb = httpVerb;
+    params.resourceType = getResourceType(url);
+    params.resourceId = getResourceIdForOffer(url);
+    return params;
+}
+
+isolated function mapResponseHeadersToObject(http:Response|http:ClientError httpResponse) returns @tainted Headers|error 
+{
+    Headers responseHeaders = {};
+    if (httpResponse is http:Response) {
+        responseHeaders.continuationHeader = getHeaderIfExist(httpResponse,CONTINUATION_HEADER);
+        responseHeaders.sessionTokenHeader = getHeaderIfExist(httpResponse,SESSION_TOKEN_HEADER);
+        responseHeaders.requestChargeHeader = getHeaderIfExist(httpResponse,REQUEST_CHARGE_HEADER);
+        responseHeaders.resourceUsageHeader = getHeaderIfExist(httpResponse,RESOURCE_USAGE_HEADER);
+        responseHeaders.itemCountHeader = getHeaderIfExist(httpResponse,ITEM_COUNT_HEADER);
+        responseHeaders.etagHeader = getHeaderIfExist(httpResponse,ETAG_HEADER);
+        responseHeaders.dateHeader = getHeaderIfExist(httpResponse,RESPONSE_DATE_HEADER);
+        return responseHeaders;
+    } else {
+        return prepareError("Error occurred while invoking the REST API");
+    }
+}
+
+isolated function mapJsonToDatabaseType([json, Headers?] jsonPayload) returns Database {
+    json payload;
+    Headers? headers;
+    [payload,headers] = jsonPayload;
+    Database db = {};
+    db.id = payload.id != ()? payload.id.toString() : EMPTY_STRING;
+    db._rid = payload._rid != ()? payload._rid.toString() : EMPTY_STRING;
+    db._self = payload._self != ()? payload._self.toString() : EMPTY_STRING;
+    if headers is Headers {
+        db["reponseHeaders"] = headers;
+    }    
+    return db;
+}
+
+isolated function mapJsonToDatabaseListType([json, Headers] jsonPayload) returns @tainted DatabaseList {
+    json payload;
+    Headers headers;
+    [payload,headers] = jsonPayload;
+    DatabaseList dbl = {};
+    dbl._rid = payload._rid != ()? payload._rid.toString() : EMPTY_STRING;
+    dbl.databases =  convertToDatabaseArray(<json[]>payload.Databases);
+    dbl.reponseHeaders = headers;
+    return dbl;
+}
+
+isolated function mapJsonToContainerType([json, Headers?] jsonPayload) returns @tainted Container {
+    json payload;
+    Headers? headers;
+    [payload,headers] = jsonPayload;
+    Container coll = {};
+    coll.id = payload.id.toString();
+    coll._rid = payload._rid != ()? payload._rid.toString() : EMPTY_STRING;
+    coll._self = payload._self != ()? payload._self.toString() : EMPTY_STRING;
+    coll.allowMaterializedViews = convertToBoolean(payload.allowMaterializedViews);
+    coll.indexingPolicy = mapJsonToIndexingPolicy(<json>payload.indexingPolicy);
+    coll.partitionKey = convertJsonToPartitionKeyType(<json>payload.partitionKey);
+    if headers is Headers {
+        coll["reponseHeaders"] = headers;
```

Comment:
Can use constant for "responseHeaders". Check all the places in this file and replace it